### PR TITLE
Fixes for NWA

### LINKED ISSUE #1057: NWA Connection Issue
Hi,

We have been testing NWA initiated NWC with Mutiny Wallet for a little while now and everything was working fine up until recently. I've noticed any new NWA connections I initiate do not seem to be working in the Mutiny wallet. 

Old connections that I have (relay.mutinywallet.com relay from 1 month ago) work perfectly. Any new connection I initiate seems to be setup correctly but then does not work. I receive a 33194 from the new wallet pubkey. NWC requests that I send are stored by your relay, but not shown in the wallet.

Here is a connection I just made that is not working:

33194 Event ID: d71ef2b40b35541664445a1766facfa01ab26976fb7639fe5eef45dc00dfcd62
23194 Event ID (pay_invoice req that was not received by wallet): 9b11439b1f084b21563a81c6a2654d152b34b9ef87210938885693e239f80280
Wallet pubkey: 51a55ebaeb2b6148c835719188bf84cbda8c1a38ce5d9f481fba52e666cc1687

Could you help me figure out what's happening?

### PR DESCRIPTION
**Pull request by benthecarman**
Cleaned up some code while I was at it. Main things that fixed the issue was broadcasting the auth event to all of our relays and not putting the relay in the auth event when it wasn't needed. 

This fixes when I was able to reproduce with zapple-pay, not sure if it fixes with nostr.wine @nostr-wine, should it send me an invoice right away?

Closes #1057

### CODE REVIEW

**Comment by nostr-wine** [comment]
Same issue testing for me. 

Connection Event ID: 21020f1b151bc9ca70477acc1440d6bc7ef37107c7f365e60208db406549fcbf
Wallet pubkey: a6f327a3c1b6ab97b891940fc7c54f90c7717af3380d200875a854a1f7590f7f

pay_invoice IDs: 0441906fc6d12fdeb6c12dac24dc7fcbe380dbc851310615e244140f45e4886b, 77b1a102178d02e647c041d29d48066a96d8b14769accde6c852ed1833cd5b2d, cb1b61d2bdd20331839e447d1b9c6e35b882f17f1e217f1df80355d7379fe8ef

File: mutiny-core/src/nostr/mod.rs [hunk 1]
```diff
@@ -124,6 +126,7 @@ impl<S: MutinyStorage> NostrManager<S> {
         // add relays for default sending
         relays.push("wss://nostr.mutinywallet.com".to_string()); // blastr
         relays.push("wss://relay.mutinywallet.com".to_string()); // strfry
+        relays.push(DEFAULT_RELAY.to_string()); // add in case it's not in the list
```
Comment (TonyGiorgio) [inline mutiny-core/src/nostr/mod.rs:129]: you're pushing the same thing twice.

**Review by TonyGiorgio — CHANGES_REQUESTED** [review]

**Comment by TonyGiorgio** [comment]
> Same issue testing for me.
> 
> Connection Event ID: 21020f1b151bc9ca70477acc1440d6bc7ef37107c7f365e60208db406549fcbf Wallet pubkey: a6f327a3c1b6ab97b891940fc7c54f90c7717af3380d200875a854a1f7590f7f
> 
> pay_invoice IDs: 0441906fc6d12fdeb6c12dac24dc7fcbe380dbc851310615e244140f45e4886b, 77b1a102178d02e647c041d29d48066a96d8b14769accde6c852ed1833cd5b2d, cb1b61d2bdd20331839e447d1b9c6e35b882f17f1e217f1df80355d7379fe8ef
> 
> EDIT: I just sent your pubkey one now as well if its helpful. ID is f38949e5d67b3d12031b5b311172a961e792ef5051da26d9963b1799e3b467fe

Nothing would have changed server side with this PR. This is on the wallet, so it wouldn't be fixed for you unless you recompiled mutiny-node and imported it into your own instance of mutiny-web. 

If @benthecarman can fix the minor PR feedback change, we can deploy this in the next few weeks for you to test again.

**Review by TonyGiorgio — APPROVED** [review]

**Comment by TonyGiorgio** [comment]
needs rebase

**Pull request merged**